### ISSUE TITLE
Sketcher: Camera doesn't always animate from the current orientation

### ISSUE THREAD

**Issue by jwueller** [FreeCAD/FreeCAD#13826]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

When in Part Design, activating a sketch brings you into Sketcher. It also animates the camera to face the plane of the sketch. However, if you then rotate the camera _within_ Sketcher, and focus the sketch again, it instantly snaps to the original camera orientation you started out with in Part Design, and then does the animation from that position, instead of where you were actually looking.

[freecad-camera-alignment-bug.webm](https://github.com/FreeCAD/FreeCAD/assets/388308/7110b319-08fe-45cb-8bf5-23233fce51f1)

I only noticed this in Sketcher so far, but this might affect other contexts too. #11234 is tangentially related, since doing camera alignment to arbitrary features probably has to deal with this at some point anyway.

### Full version info

```shell
OS: Arch Linux
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.36996 (Git) AppImage
Build type: Release
Branch: main
Hash: d92f74ef019fcedbdcdc32055ab2437c950ac23b
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Installed mods: 
  * OpenDark 2024.3.13
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
So the issue is, that there is no smooth animation? Is this also the case when you right click in the active sketch and select view sketch?

**Comment by jwueller** [comment]
No, the issue is that the animation sometimes starts from the wrong position, which suggests it's using stale data. The animation itself is perfectly smooth. Yes, this also happens if I re-view the sketch from the context menu.

**Comment by maxwxyz** [comment]
@Rexbas you are the guy for this 😄

**Comment by Rexbas** [comment]
Yes, I will look at it.

**Comment by Rexbas** [comment]
I looked into it and currently, when you exit the sketch mode, the camera is restored to its orientation before you entered the sketch mode, without animations.

The problem is that when you double click on the sketch in the tree view then the sketch mode is exited and entered again. This means it restores the camera first, then it starts an animation to the sketch plane again. I do not know how to check if a sketch is refocused by double clicking in the tree view or if a sketch is closed. Thus, I have no way to prevent the restore code from running.

However, it is possible to disable the camera restore completely using the parameter editor (Tools -> Edit parameters...). Set `RestoreCamera` to false in `BaseApp/Preferences/Mod/Sketcher/General`. I personally would not mind if the default is false because I think the restore is disorienting. But others may disagree.

**Comment by jwueller** [comment]
Curious, any idea why the sketch even closes when you (re)activate it in the tree? Or is the tree just hard-coded to always exit whichever mode the viewport is in and then enter the clicked one, without regard for whether it's already active?

**Comment by Rexbas** [comment]
It indeed exits whatever is active first, regardless of what is active. However, this code is not specific to the tree view: https://github.com/FreeCAD/FreeCAD/blob/7d738f161a64b1fc154abed244603a89b3439617/src/Gui/Document.cpp#L296-L301

**Comment by maxwxyz** [comment]
the entire double click in the tree is a mess..  
* https://github.com/FreeCAD/FreeCAD/issues/12839

**Comment by luzpaz** [comment]
re-opening since it's not clear this issue is fixed.

**Comment by FEA-eng** [comment]
@Rexbas Still relevant?

**Comment by Rexbas** [comment]
Yes still relevant.

**Comment by FEA-eng** [comment]
Still relevant

```
OS: Windows 10 build 19045
Architecture: x86_64
Version: 1.1.0dev.20251111 (Git shallow)
Build date: 2025/11/11 16:45:10
Build type: Release
Branch: main
Hash: 4fbb389ce1295e04a5ad7f3a63f68b63b4f4036c
Python 3.11.14, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, IfcOpenShell 0.8.2, OCC 7.8.1
Locale: Polish/Poland (pl_PL)
Navigation Style/Orbit Style/Rotation Mode: CAD/Rounded Arcball/Window center
Stylesheet/Theme/QtStyle: unset/FreeCAD Light/fusion
Logical DPI/Physical DPI/Pixel Ratio: 96/90.0633/1
```

**Comment by FEA-eng** [comment]
Still relevant

```
OS: Windows 10 build 19045
Architecture: x86_64
Version: 1.2.0dev.20260526 (Git shallow)
Build date: 2026/05/26 17:25:08
Build type: Release
Branch: (HEAD detached at 26ca5b4bd)
Hash: 26ca5b4bda657d359a783e9de99d7ae0a07fc513
Python 3.11.15, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, Clipper2 , IfcOpenShell 0.8.2, OCC 7.8.1
Locale: Polish/Poland (pl_PL) [ OS: Polish/Poland (pl_PL) ]
Stylesheet/Theme/QtStyle: FreeCAD.qss/FreeCAD Light/
Navigation Style/Orbit Style/Rotation Mode: Blender/Rounded Arcball/Window center
Logical DPI/Physical DPI/Pixel Ratio: 96/90.0633/1
```
